src/plugins/intel_gpu/src/graph/impls/onednn/utils.cpp

```diff
@@ -297,150 +333,264 @@ static const std::map<cldnn::format::type, dnnl::memory::format_tag> format_map_
-        dims.push_back(l.batch());
-        dims.push_back(l.get_tensor().count() / l.batch());
-    } else if (target_fmt == dnnl::memory::format_tag::abc) {
-        dims.push_back(l.batch());
-        dims.push_back(l.feature());
-        dims.push_back(l.spatial(1));
-    } else if (target_fmt == dnnl::memory::format_tag::acb) {
-        dims.push_back(l.batch());
-        dims.push_back(l.spatial(1));
-        dims.push_back(l.feature());
-    } else if (target_fmt == dnnl::memory::format_tag::abdc) {
-        dims.push_back(l.batch());
-        dims.push_back(l.feature());
-        dims.push_back(l.spatial(0));
-        dims.push_back(l.spatial(1));
-    } else if (target_fmt == dnnl::memory::format_tag::abced) {
-        dims.push_back(l.batch());
-        dims.push_back(l.feature());
-        dims.push_back(l.spatial(2));
-        dims.push_back(l.spatial(0));
-        dims.push_back(l.spatial(1));
-    } else if (target_fmt == dnnl::memory::format_tag::abcdfe) {
-        dims.push_back(l.batch());
-        dims.push_back(l.feature());
-        dims.push_back(l.spatial(3));
-        dims.push_back(l.spatial(2));
-        dims.push_back(l.spatial(0));
-        dims.push_back(l.spatial(1));
-    } else if (target_fmt == dnnl::memory::format_tag::ba) {
-        dims.push_back(l.feature());
-        dims.push_back(l.get_tensor().count() / l.feature());
-    } else if (flatten) {
-        dims = flatten_tensor(l.get_tensor());
-    } else {
-        // clDNN expresses 3d tensor with 4d format. This code is to use 3d format on oneDNN for such case.
-        // However, if the memory::desc to be converted is related to another blocked format, it should be expanded to a 4d tensor.
-        auto shape_rank = l.is_dynamic() ?
-            static_cast<size_t>(l.get_partial_shape().rank().get_length()) : l.get_shape().size();
-        if (shape_rank == 3 && !need_blocked && !is_grouped) {
-            dims.push_back(l.batch());
-            dims.push_back(l.feature());
-            // In cldnn::layer, when it is a 3D shape, the values ​​of the XY axes can sometimes be flipped,
-            // so the larger value of the two is used.
-            dims.push_back(std::max(l.spatial(0), l.spatial(1)));
-            auto it = format_map_cldnn_4d_to_onednn_3d.find(l.format);
-            if (it != format_map_cldnn_4d_to_onednn_3d.end()) {
-                target_fmt = it->second;
+    // Pre-calculate shape rank once in constructor
+    static size_t calculate_shape_rank(const cldnn::layout& l) {
+        return l.is_dynamic() ?
+            static_cast<size_t>(l.get_partial_shape().rank().get_length()) :
+            l.get_shape().size();
+    }
+
+public:
+    explicit MemoryDescriptorBuilder(const cldnn::layout& l, dnnl::memory::format_tag target_fmt = dnnl::memory::format_tag::undef)
+        : _layout(l)
+        , _target_fmt(target_fmt)
+        , _shape_rank(calculate_shape_rank(l))  // Calculate once
```

Comment:
Not sure but using `static_cast<size_t>(l.get_partial_shape().rank().get_length())` is enough instead `calculate_shape_rank`?

Reply:
Partial shape is checked only for dynamic shape, so it cannot be removed.